augur/util_support/date_disambiguator.py

```diff
@@ -60,7 +60,32 @@ def range(self):
             resolve_uncertain_int(self.uncertain_date_components["m"], "max"),
             resolve_uncertain_int(self.uncertain_date_components["d"], "max"),
         )
-        max_date = min(max_date, datetime.date.today())
+
+        # Set default bounds.
+        lower_bound = None
+        upper_bound = datetime.date.today()
+
+        # TODO: Allow more granularity than year alone.
+        if self.min_max_year:
+            lower_bound = datetime.date(int(self.min_max_year[0]), 1, 1)
+
+            if len(self.min_max_year) == 2:
+                upper_bound = datetime.date(int(self.min_max_year[1]), 12, 31)
```

Comment:
The code below makes the assumption that `lower_bound < upper_bound`, so we should check that's true either by a) explicitly checking it after we process user input, or b) verify the user input is correctly ordered before processing it.

i.e. a user could provide `--year-bounds 2022 2018` and we should handle that.

Reply:
Done: c4d16597e4a67904bcc43da73a4fae5e10749e34

I chose (a) and bundled it with some other error handling which is nice to have upon user input like this.